catalog/CatalogRelation.hpp

```diff
@@ -203,35 +233,70 @@ class CatalogRelation : public CatalogRelationSchema {
   }
 
   /**
-   * @brief Check if an index by the given name exists already
-   *
-   * TODO (ssaurabh): Provide a better interface to query indices.
+   * @brief Check whether an index with the given exists or not.
    *
-   * @param The query index name
-   * @return true, if the index with the given name exists, otherwise false.
+   * @param index_name Name of the index to be checked.
+   * @return Whether the index exists or not.
    **/
-  bool hasIndexWithName(const std::string& name) {
-    SpinSharedMutexExclusiveLock<false> lock(indices_mutex);
-    std::vector<std::string>::iterator it = std::find(indices_.begin(), indices_.end(), name);
-    if (it != indices_.end()) {
-      return true;
-    } else {
-      return false;
-    }
+  bool hasIndexWithName(const std::string &index_name) const {
+    SpinSharedMutexExclusiveLock<false> lock(index_scheme_mutex);
```

Comment:
Change to `SpinSharedMutexSharedLock` for readers.

FYI, `SpinSharedMutexExclusiveLock` is used for writers.